zoo/analytics/tasks/utils.py

```diff
@@ -59,3 +61,64 @@ def resolve_unknown_format(version: str) -> str:
         else:
             break
     return ".".join(temp_version)
+
+
+class DockerImageId:
+    """Represents parsed docker image id.
+
+    TODO: move this to some more appropriate OSS package.
+    """
+
+    # pylint: disable=too-many-instance-attributes # Ignore PyCommentedCodeBear
+
+    def __init__(self, image: str):  # Ignore RadonBear
+        self.full_image_name = image
+
+        self.registry = None
```

Comment:
maybe just a nitpick, but wouldn't it be better to make this properties? it would help readability - you would read just few lines of code for particular attribute parsing. Also "go to definition" would be easier when you see it in some code as it would point you directly to code parsing it. 

properties could be cached. dunno if guys are using already something, but there are multiple implementation we do not need to do it.

Reply:
Makes sense, although changing it made the parser much longer, its easier to understand and it will be much easier to maintain it.